### PR TITLE
model_checkpoint to save all models

### PR DESCRIPTION
**Pull request by williamFalcon**
Finishes #1160

### CODE REVIEW

**Comment by williamFalcon** [comment]
@jeremyjordan @MattPainter01 you guys wanted more changes to this?  i’ll approve. when you guys are happy you already have my vote to merge.

**Comment by williamFalcon** [comment]
false lol. can’t self approve

**Comment by Borda** [comment]
> false lol. can’t self approve

:-D

File: pytorch_lightning/callbacks/model_checkpoint.py [hunk 1]
```diff
@@ -216,7 +216,8 @@ def _do_check_save(self, filepath, current, epoch):
         if len(self.best_k_models) == self.save_top_k:
             delpath = self.kth_best_model
             self.best_k_models.pop(self.kth_best_model)
-            self._del_model(delpath)
+            if self.save_top_k > 0:
```
Comment (Borda) [inline pytorch_lightning/callbacks/model_checkpoint.py:219]: @jeremyjordan this if shall be one level up, you don't want to pop (remove from list) model which you don't delete, right?

**Pull request merged**